hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java

```diff
@@ -422,6 +437,10 @@ public boolean isPersistWanReplicatedData() {
         return persistWanReplicatedData;
     }
 
+    public RecordStore<?> getExistingRecordStore(int partitionId) {
```

Comment:
`map.getMapServiceContext().getExistingRecordStore(partitionId, map.getName())`? It might be ugly, but otherwise we will duplicate all `MapServiceContext` methods in `MapContainer` sooner or later.

Reply:
It's just a convenient shortcut. What is the harm of having it?

Reply:
Because sooner or later that would lead to all `MapServiceContext` methods duplicated in `MapContainer`. Currently, there are no methods in `MapContainer` working with record stores, all such methods are located in `MapServiceContext`.

Reply:
`getIndexes` and `getIndexDefinitions` do almost the same - get partition container, then some field from it. I honestly do not see any problem with it. It just provides a convenience instead of passing the name form the outside.

Reply:
Indexes is a different story, global ones are managed by `MapContainer` while stores are not managed by `MapContainer`. I do see a problem, but it's up to you.

Reply:
Fixed.